### PR TITLE
Improve flow of Introduction and a few adjustments with regard to the OHBM abstract

### PR DESCRIPTION
**Pull request by adswa**
This introduces changes to the first parts of the book:

- I'm removing the alpha stage note on the front page, and replace it with a short introduction of the aims of the handbook as stated in our abstract
- A few tweaks to some introductory sections to connect them better, define the scope, a bit of restructuring, ...
- Adding a ``Teaching with the DataLad Handbook`` section that points to the educational resources for trainers we provide

### CODE REVIEW

File: docs/teaching.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+attendees of every career stage (MSc students up to PIs). The sessions are either
+part of a lecture series (with bi-weekly 90 minute sessions) or workshops of different
+lengths. Sessions in the lecture series are based on each chapter. Longer workshops
+combine several chapters. You can find the slides for the workshops in the
+`companion course repository <https://github.com/datalad-handbook/course>`_.
+Slides are made using `reveal.js <https://github.com/hakimel/reveal.js/>`_.
+They are available as PDFs in ``talks/PDFs/``, or as the source ``html`` files
+in ``talks/``.
+
+Enhance talks and workshops with code casts
```
Comment (mih) [inline docs/teaching.rst]: I still don't think that 'cast' is a term that can be intuitively understood. I think "demo" is a better choice.

File: docs/teaching.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+"""""""""""""""""""""""""""""""""""""""""""
+
+Any number of code snippets in the handbook that are created with the ``runrecord``
+directive can be aggregated into a series of commands that can be sequentially
+executed as a code cast using the
```
Comment (mih) [inline docs/teaching.rst]: as a code cast -> as a demo

**Comment by adswa** [comment]
Good point, I have changed this.

**Comment by adswa** [comment]
I'll merge this in a couple of minutes.

**Pull request merged**